activerecord/lib/active_record/migration.rb

```diff
@@ -1197,12 +1197,11 @@ def migrate_without_lock
           raise UnknownMigrationVersionError.new(@target_version)
         end
 
-        result = runnable.each do |migration|
+        record_environment
+
+        runnable.each do |migration|
           execute_migration_in_transaction(migration, @direction)
         end
-
-        record_environment
```

Comment:
This leads me to believe that there was a reason why we recorded the environment data after migrations get ran but I can't think of one. 🤔